scripts/setup_ubuntu.sh

```diff
@@ -49,25 +51,39 @@ run_as_original_user() {
 # Add custom repositories
 #
 curl -fL https://apt.kitware.com/keys/kitware-archive-latest.asc | apt-key add -
-apt-add-repository -y $NO_UPDATE "deb https://apt.kitware.com/ubuntu/ ${RELEASE} main"
+if [ "$VERSION" == "21" ]; then
+    # no hirsute aka 21.04 apt repo yet, use focal one instead
+    apt-add-repository -y $NO_UPDATE "deb https://apt.kitware.com/ubuntu/ focal main"
+else
+    apt-add-repository -y $NO_UPDATE "deb https://apt.kitware.com/ubuntu/ ${RELEASE_CODENAME} main"
+fi
 
 add-apt-repository -y $NO_UPDATE ppa:git-core/ppa
 
 # Clang isn't present in the xenial repos so we need to add the repo
-VERSION=$(lsb_release --release --short | cut -d . -f 1)
-if [ "$VERSION" == "16" ]; then
+# Clang-10 isn't present in the hirsute repos any more, so we need to add the upstream repo
+if [[ "$VERSION" == "16" ||  "$VERSION" == "21" ]]; then
   curl -fL https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add -
-  apt-add-repository -y $NO_UPDATE "deb http://apt.llvm.org/${RELEASE}/ llvm-toolchain-${RELEASE}-10 main"
+  apt-add-repository -y $NO_UPDATE "deb http://apt.llvm.org/${RELEASE_CODENAME}/ llvm-toolchain-${RELEASE_CODENAME}-10 main"
 fi
 
 if [[ -n "${SETUP_TOOLCHAIN_VARIANTS}" ]]; then
   apt-add-repository -y $NO_UPDATE ppa:ubuntu-toolchain-r/test
 fi
 
-RELEASE_ID="$(lsb_release --id --short | tr 'A-Z' 'a-z')"
-RELEASE_CODENAME="$(lsb_release --codename --short)"
-curl "https://apache.jfrog.io/artifactory/arrow/$RELEASE_ID/apache-arrow-apt-source-latest-$RELEASE_CODENAME.deb" \
-  --output /tmp/apache-arrow-apt-source-latest.deb
+
+if [ "$VERSION" == "21" ]; then
+    # no hirsute aka 21.04 release of apache arrow yet, use focal one instead
+    # we must also get libre2-5 from focal
+    curl "https://apache.jfrog.io/artifactory/arrow/$RELEASE_ID/apache-arrow-apt-source-latest-focal.deb" \
+         --output /tmp/apache-arrow-apt-source-latest.deb
+    wget "mirrors.kernel.org/ubuntu/pool/main/r/re2/libre2-5_20200101+dfsg-1build1_amd64.deb" \
+         -O /tmp/libre2-5.deb
```

Comment:
Also don't use a mirror. Use the main ubuntu repo for this.

Reply:
funny enough, I actually used wget because 
```
curl "http://mirrors.kernel.org/ubuntu/pool/main/r/re2/libre2-5_20200101+dfsg-1build1_amd64.deb" \
         --output /tmp/libre2-5.deb
```
results in a 0 byte file, whereas `wget` just worked

that said, changing to the main ubuntu repo 
```
curl "http://archive.ubuntu.com/ubuntu/pool/main/r/re2/libre2-5_20200101+dfsg-1build1_amd64.deb" --output libre2-5.deb```
fixes that issue, so I guess we get two birds with one stone.